performance-tests/index.mjs

```diff
@@ -0,0 +1,141 @@
+// @flow strict
+// import url from 'url';
+import playwright from 'playwright';
+import { createWorker, PSM } from 'tesseract.js';
+import waitForRenderedItems from './waitForRenderedItems.mjs';
+
+const testURL = new URL('http://localhost:8888/integration-test/masonry');
+const DEFAULT_SCROLL_COUNT = 5;
+const PINS_COUNT = 5;
+const PAGE_HEIGHT = 1000;
+
+/**
+ * Args:
+ * --scroll-count: number
+ *  Number of times to scroll the page and fetch items
+ * --debug: flag
+ *  Open browser with dev tools, more verbose logging, and don't close browser when done
+ * --verbosity: 'silent' | 'default' | 'verbose'
+ * Logging verbosity
+ * --two-col-items
+ * Enable two column items
+ * --realistic-pin-heights
+ * Enable realistic pin heights
+ */
+(async () => {
+  // Get args
+  const args = process.argv.slice(2);
+
+  if (args.find((arg) => arg.includes('--two-col-items'))) {
+    testURL.searchParams.append('twoColItems', '1');
+  }
+  if (args.find((arg) => arg.includes('--realistic-pin-heights'))) {
+    testURL.searchParams.append('realisticPinHeights', '1');
+  }
+
+  const scrollCountArg =
+    args.find((arg) => arg.startsWith('--scroll-count='))?.split('=')[1] ??
+    DEFAULT_SCROLL_COUNT;
+  const scrollCount = parseInt(scrollCountArg, 10);
+
+  const debug = args.includes('--debug');
+
+  const verbosityArg = args
+    .find((arg) => arg.startsWith('--verbosity='))
+    ?.split('=')[1];
+
+  const launchOptions = debug
+    ? { headless: false, devtools: false }
+    : { headless: true };
+
+  const log = (message /*: string */) =>
+    ['default', 'verbose'].includes(verbosityArg)
+      ? console.log(message)
+      : undefined;
+
+  // Launch browser and page
+  const browser = await playwright.chromium.launch(launchOptions);
+  const page = await browser.newPage();
+
+  const client = await page.context().newCDPSession(page);
+  await client.send('Overlay.setShowFPSCounter', {
+    show: true,
```

Comment:
this is an interesting approach. this certainly works and gets us the average FPS (although in testing locally, i was running into issues with the OCR actually getting data from the screenshot)

there are a few downsides with this approach imo:

- we are limited to average FPS (vs being able to get data on longest frame, shortest frame, etc)
- this will always be the average FPS of the entire session
- we have to do this weird dance of capturing a screenshot, saving it, reading it using via ocr library

Alternatively, we could hook into Chrome's tracing information (i.e. the data you get when you do a performance trace) since this includes information about frames being drawn as well as the timestamps for when the event happens.

Was playing around with the CDP protocol and we should be able to get this information directly through the [Tracing](https://chromedevtools.github.io/devtools-protocol/tot/Tracing/) API. something like
```
const session = await page.context().newCDPSession(page);
// start tracing
await session.send('Tracing.start');
//scroll scroll scroll
// listen for trace events
session.on('Tracing.dataCollected', (data) => {
  const events = data.value.filter((e) => {
    // i think the event we want here is `DrawFrame`
    return e.name === 'DrawFrame'
});
//end tracing
await session.send('Tracing.end');
```

The general idea here is that we should be able to take this data and use that to derive data such as longest frame, avg fps, etc

Let me know what you think about this approach